HotelManagement/src/core/Commodities/AbstractCommodity.java

```diff
@@ -0,0 +1,33 @@
+package core.Commodities;
+
+public abstract class AbstractCommodity {
+	private int inventoryNumber;
+
+	public AbstractCommodity(int inventoryNumber) {
+		this.setInventoryNumber(inventoryNumber);
+	}
+
+	public abstract void prepare();
+
+	public int getInventoryNumber() {
+		return inventoryNumber;
+	}
+
+	public void setInventoryNumber(int inventoryNumber) {
+		if (inventoryNumber < 0) {
+			throw new IllegalArgumentException("Inventory number cannot be negative.");
+		}
+
+		this.inventoryNumber = inventoryNumber;
+	}
+
+	@Override
+	public int hashCode() {
+		return inventoryNumber;
+	}
+
+	@Override
+	public boolean equals(Object obj) {
```

Comment:
use instanceof operator to check if obj is of class AbstractCommodity or else it is liable to ClassCastExceptions.